src/mongo-utils.ts

```diff
@@ -16,10 +17,39 @@ export class MongoUtils {
 	): Promise<mongodb.Db> {
 		const log = global.log.module('mongo');
 		log.info(`Connecting to ${MongoUtils.hidePasswordFromURI(url)}...`);
-		global.dbClient = await mongodb.MongoClient.connect(url, options);
-		const db = (global.dbClient as mongodb.MongoClient).db();
+
+		const mongoClient = new mongodb.MongoClient(url, options);
+
+		mongoClient.on('serverOpening', () => {
+			log.info(`-- mongo-utils.ts serverOpening --`);
+		});
+		mongoClient.on('open', () => {
+			log.info(`-- mongo-utils.ts open --`);
+		});
+		mongoClient.on('close', () => {
+			log.info(`-- mongo-utils.ts close --`);
+		});
+		mongoClient.on('serverClosed', () => {
+			log.info(`-- mongo-utils.ts serverClosed --`);
+		});
+		mongoClient.on('reconnect', () => {
+			log.info(`-- mongo-utils.ts reconnect --`);
+		});
+		mongoClient.on('timeout', (e) => {
+			log.warn(`-- mongo-utils.ts timeout --`, e);
+		});
+
+		const dbClient = await mongoClient.connect();
+		global.dbClient = dbClient;
+		const db = dbClient.db();
 		global.db = db;
+
+		(db as any).s.topology.on('reconnectFailed', () => {
+			log.warn(`-- mongo-utils.ts reconnect failed --`);
```

Comment:
```suggestion
			log.warn(`MongoDB Client reconnect failed.`);
                        process.kill(process.pid, "SIGTERM");
```

if reconnect failed, we have to shut down

Reply:
check how is it done in n9-node-routing ;)

Reply:
@BenjD90 done!